### ISSUE TITLE
Add indentation matching support

### ISSUE THREAD

**Issue by rvantonder** [comby-tools/comby#65]
For layout or indentation-sensitive languages like Python, enforce the constraint that matching bodies should be (relatively) indented for, e.g., `def`, `if`, `class`, etc.

**Comment by isovector** [comment]
Relatedly, I'd like a way to match a given indentation level, in Haskell. For example, I'd like to match the following:

```haskell
instance Eq E where
  E x /= E y =
    let foo = x /= y
     in foo
```

with the config:

```toml
[only-neq]

match='''
instance :[ctx]Eq :[name] where
  :[x] /= :[y] = :[z\n]
'''

rewrite='''
instance :[ctx]Eq :[name] where
  :[x] == :[y] = not $ :[z]
'''
```

But the diff is:

```diff
 instance Eq E where
-  E x /= E y =
-    let foo = x /= y
-     in foo
+  E x == E y = not $ let foo = x /= y
+
+in foo
```

**Comment by rvantonder** [comment]
@isovector yeah, agree, that'd be nice to have. There's a janky workaround with Python actually using `pyindent` but no equivalent for Haskell as far as I know. The grammar/syntax for python makes the problem fairly approachable, but I'm unsure if I can use similar syntactic cues for Haskell (might be harder, I'd have to investigate).

About that specific example, you can preserve leading whitespace at the line level by capturing it with a regular expression. It's ugly and I'm entirely unsatisfied with the ergonomics of this, but just sharing the possibility in case you're desperate to get some more mileage with this issue: https://bit.ly/3DGmptU

**Comment by isovector** [comment]
Thanks @rvantonder, that's already a significant improvement. Cheers!
